src/lib-voyager.tsx

```diff
@@ -0,0 +1,108 @@
+// This module is intended to be used when embedding voyager
+// in some other context than the orgiginal app.
+//
+// It provides factory methods for creating instances of the Voyager application
+// and should eventually also export a react component if one is doing that kind
+// of integration.
+
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import {Store} from 'redux';
+
+import 'font-awesome-sass-loader'; // TODO should this move to App?
+import {StateWithHistory} from 'redux-undo';
+import {isString} from 'vega-lite/build/src/util';
+import {App, VoyagerConfig, VoyagerData} from './components/app';
+import {StateBase} from './models/index';
+import {configureStore} from './store';
+
+type Container = string | HTMLElement;
+
+
+/**
+ * The Voyager class encapsulates the voyager application and allows for easy
+ * instantiation and interaction from non-react projects.
+ */
+class Voyager {
+  private container: HTMLElement;
+  private config: VoyagerConfig;
+  private data: VoyagerData;
+  private store: Store<StateWithHistory<Readonly<StateBase>>>;
+
+  constructor(container: Container, config: VoyagerConfig, data: VoyagerData) {
+    if (isString(container)) {
+      this.container = document.querySelector(container) as HTMLElement;
+      // TODO throw error if not found
+    } else {
+      this.container = container;
+    }
+    this.init();
+  }
+
+  /**
+   * Update the dataset currently loaded into voyager
+   *
+   * @param {VoyagerData} data
+   *
+   * @memberof Voyager
+   */
+  public updateData(data: VoyagerData) {
+    this.data = data;
+    this.render(data, this.config);
+  }
+
+  /**
+   * Update the configuration of the voyager application.
+   *
+   * @param {VoyagerConfig} config
+   *
+   * @memberof Voyager
+   */
+  public updateConfig(config: VoyagerConfig) {
+    this.config = config;
+    this.render(this.data, config);
```

Comment:
Yes I think it should re-render the whole app, that way anything that depends on these props (or things derived from them) in the store will update accordingly. I might be missing a nuance to your question so let me know. However maybe i should add that calling ReactDOM.render in the code below is just like rendering a component further down the state tree, react will do dom-diffing, and components will run their regular lifecycle methods etc, it wont 'force' a fresh render so to speak (e.g. removing the current dom elements completely and starting from scratch).